### PR TITLE
Json Editor Flat Schema Builder

### PR DESCRIPTION
**Pull request by duredhel13**
JSON Editor Schema Builder Mode based on the new `ngx-json-editor-flat` interface.

Designs:
- https://www.figma.com/file/fGr0PYeAiHLTDjqCaoIOI6/Plugin-IDE?node-id=206%3A0

Usage: 
```
jsonEditorSchemaBuilderModel: any = {};

customFormats = ['password', 'code', 'date', 'date-time', 'custom'];

<ngx-json-editor-flat
      [(model)]="jsonEditorSchemaBuilderModel"
      [schema]="{}"
      label="Schema"
      [typeCheckOverrides]="typeOverrides"
      [schemaBuilderMode]="true"
      [formats]="customFormats"
      (schemaChange)="schemaChange($event)">
 </ngx-json-editor-flat>
```

Couple of things missing from this implementation and will be added as separate PRs:
- The ability to add examples to an object property.
- The ability for an object property to have multiple types (currently only one is supported).

### CODE REVIEW

**Comment by Hypercubed** [comment]
I belive the component, as-is, is not OnPush compatible and will not work well on pages that are OnPush.

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor/json-editor-node/node-types/object-node/object-node.component.html [hunk 1]
```diff
@@ -1,29 +1,29 @@
 <div [hidden]="!expanded">
-  <div *ngFor="let prop of propertyIndex | keyvalue; trackBy: trackBy">
+  <div *ngFor="let prop of getProperties(); trackBy: trackBy">
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor/json-editor-node/node-types/object-node/object-node.component.html]: I think the pipe was better here.... this is going to run on each "digest".

Comment (duredhel13) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor/json-editor-node/node-types/object-node/object-node.component.html]: The pipe was messing with the order of properties. Was listing props by alphabetic order.

Comment (Hypercubed) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor/json-editor-node/node-types/object-node/object-node.component.html]: I'd suggest... if possible... create your own pipe.  Would be better than repeatedly calling `getProperties`... which returns a new array each time.

Comment (duredhel13) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor/json-editor-node/node-types/object-node/object-node.component.html]: Done

**Pull request merged**